Доработано

MikhailFrantsuzov opened the pull request.

MikhailFrantsuzov committed:
Доработано

MikhailFrantsuzov commented:
Изначально я закомментил цвета потому, что так посоветовал наставник. У меня в проекте возникает конфликт между ними и цветами в ресурсах в Assets.xcassets.
Потом я их раскомментил, но решил поменять имена, чтобы не было конфликта.
У меня проект запускается в обоих случаях, и если их закомментить, и если поменять названия.
Другого решения этого вопроса, кроме как удалить их либо в ресурсах, либо в расширении мне не предложили.
Ок. Я грохнул цвета в ресурсах (надеюсь это не повлияет на проект в дальнейшем) и вернул расширение с ними к первоначальному виду. Надеюсь теперь всё будет работать без конфликтов. У меня проект запускается.
![Uploading Снимок экрана 2023-11-18 в 16.33.11.png…]()

DanilaG commented:
Привет!
После разъяснительного комментария, я понял что проблема в разных версиях XCode (у тебя более новый, который генерирует сам extension для цветов). Вижу 2 пути возможного решения:

- Вернуть цвета Яндекса в Assets. Воспользоваться в коде тем extension что написал ты. Т. е. везде где используется `ypGreen` использовать `ypGreenColor`, везде где `ypRed` использовать `ypRedColor` и так далее. Тогда, не будет возникать конфликтов с автосгенерированным кодом + код будет компилироваться на старых версиях XCode

- Вернуть цвета Яндекса в Assets. Закомментировать цвета, как это было изночально, но добавить объяснение о различии xcode версий. Тогда, файл `UIColor+Extensions.swift` примет следующий вид. Но при этом, вижу вероятность что новый ревьювер может не заметить комментарий и проблема повторно воспроизведётся

```
// Использую XCode с генерацией цветов, в extension не нуждаюсь
extension UIColor {
//    static var ypGreen: UIColor { UIColor(named: "YP Green") ?? UIColor.green }
//    static var ypRed: UIColor { UIColor(named: "YP Red") ?? UIColor.red }
//    static var ypBlack: UIColor { UIColor(named: "YP Black") ?? UIColor.black}
//    static var ypBackground: UIColor { UIColor(named: "YP Background") ?? UIColor.darkGray }
//    static var ypGray: UIColor { UIColor(named: "YP Gray") ?? UIColor.gray }
//    static var ypWhite: UIColor { UIColor(named: "YP White") ?? UIColor.white}
}
```

В любом случае, давай вернём цвета Яндекса в Assets и проверим, что теперь используются именно они.

Если возникнут вопросы, обязательно задавай их Наставнику!

MikhailFrantsuzov committed:
Вернул цвета Яндекса в Assets. Закомментировал цвета в UIColor+Extensions и оставил разъяснение.

MikhailFrantsuzov commented:
Вернул цвета Яндекса в Assets. Закомментировал цвета в UIColor+Extensions и оставил разъяснение.

MikhailFrantsuzov committed:
Задание спринт № 6.
1. Приложение запускается и работает корректно по функциональным и техническим требованиям.
2. Добавлена работа с сетью (фильмы для квиза должны загружаться из открытого API IMDb).
3. В приложении использованы фильмы из топ-250 и самых популярных фильмов IMDb.
4. Обновлён метод viewDidLoad() в MovieQuizViewController — там вы должны начинать загрузку данных и показывать индикатор загрузки (уже показанный метод showLoadingIndicator()).
5. Если происходит ошибка загрузки данных о фильмах, то показывается алерт с ошибкой.
6. Приложение не пишет в логи сообщения об ошибках, связанных с работой с UI из другого потока.

MikhailFrantsuzov committed:
Задание по проекту спринта № 6.
1. Приложение запускается и работает корректно по функциональным и техническим требованиям.
2. Добавлена работа с сетью (фильмы для квиза должны загружаться из открытого API IMDb).
3. В приложении использованы фильмы из топ-250 и самых популярных фильмов IMDb.
4. Обновлён метод viewDidLoad() в MovieQuizViewController — там вы должны начинать загрузку данных и показывать индикатор загрузки (уже показанный метод showLoadingIndicator()).
5. Если происходит ошибка загрузки данных о фильмах, то показывается алерт с ошибкой.
6. Приложение не пишет в логи сообщения об ошибках, связанных с работой с UI из другого потока.
